src/HotChocolate/Core/src/Types/Types/Relay/NodeResolverTypeInterceptor.cs

```diff
@@ -172,12 +190,16 @@ public override void OnAfterMergeTypeExtensions()
 
                 // For the id field we need to make sure that a result formatter is registered
                 // that encodes the IDs returned from the id field.
-                RelayIdFieldHelpers.ApplyIdToField(idDef);
+                if (idDef is ObjectFieldDefinition objectIdDef)
+                {
+                    // TODO: For interface types ?!
```

Comment:
// TODO: For interface types ?!

Reply:
`RelayIdFieldHelpers.ApplyIdToField` rely heavy on `ObjectFieldDefinition`, should we have another implementation for `InterfaceFieldDefinition` or there is another way to solve this?

Reply:
OK ... dug into the code ... this is applying formatters for the execution. The interfaces does not have a resolver. What we need to do is getting all implementing fields and apply it there.

Reply:
There is another assumption we should introduce, for a node resolver returning an interface ... I think even a union would work... the following must become true:
- all implementations aka possible types must become nodes.